core/types/transaction.go

```diff
@@ -366,27 +366,28 @@ func (tx *Transaction) Cost() *big.Int {
 	return total
 }
 
-// RollupDataGas is the amount of gas it takes to confirm the tx on L1 as a rollup
-func (tx *Transaction) RollupDataGas() RollupGasData {
+// L1CostData caches the information needed to efficiently compute the L1 data fee even if the L1
+// cost function changes.
+func (tx *Transaction) L1CostData() L1CostData {
 	if tx.Type() == DepositTxType {
-		return RollupGasData{}
+		return L1CostData{}
 	}
-	if v := tx.rollupGas.Load(); v != nil {
-		return v.(RollupGasData)
+	if v := tx.l1CostData.Load(); v != nil {
+		return v.(L1CostData)
 	}
 	data, err := tx.MarshalBinary()
 	if err != nil { // Silent error, invalid txs will not be marshalled/unmarshalled for batch submission anyway.
 		log.Error("failed to encode tx for L1 cost computation", "err", err)
 	}
-	var out RollupGasData
-	for _, byt := range data {
-		if byt == 0 {
-			out.Zeroes++
+	out := L1CostData{}
```

Comment:
nit: can we switch this to a `NewL1CostData(data)` method that does the counting internally?

Reply:
nice, done.

Reply:
nice, done.

Reply:
idea from https://github.com/ethereum-optimism/op-geth/pull/202#discussion_r1428167906, shoutout @tynes